src/android/com/microsoftopentechnologies/intellij/wizards/createvm/MachineSettingsStep.java

```diff
@@ -0,0 +1,219 @@
+
+        if (name.length() > 15 || name.length() < 3) {
+            JOptionPane.showMessageDialog(null, "Invalid virtual machine name. The name must be between 3 and 15 character long.", "Error creating the virtual machine", JOptionPane.ERROR_MESSAGE);
+            return this;
+        }
+
+        if (!name.matches("^[A-Za-z][A-Za-z0-9-]+[A-Za-z0-9]$")) {
+            JOptionPane.showMessageDialog(null, "Invalid virtual machine name. The name must start with a letter, \n" +
+                    "contain only letters, numbers, and hyphens, " +
+                    "and end with a letter or number.", "Error creating the virtual machine", JOptionPane.ERROR_MESSAGE);
+            return this;
+        }
+
+        if (!pass.equals(conf)) {
+            JOptionPane.showMessageDialog(null, "Password confirmation should match password", "Error creating the service", JOptionPane.ERROR_MESSAGE);
+            return this;
+        }
+
+        if (!pass.matches("(?=^.{8,255}$)((?=.*\\d)(?=.*[A-Z])(?=.*[a-z])|(?=.*\\d)(?=.*[^A-Za-z0-9])(?=.*[a-z])|(?=.*[^A-Za-z0-9])(?=.*[A-Z])(?=.*[a-z])|(?=.*\\d)(?=.*[A-Z])(?=.*[^A-Za-z0-9]))^.*")) {
+            JOptionPane.showMessageDialog(null, "The password does not conform to complexity requirements.\n" +
+                    "It should be at least eight characters long and contain a mixture of upper case, lower case, digits and symbols.", "Error creating the virtual machine", JOptionPane.ERROR_MESSAGE);
+            return this;
+        }
+
+
+        model.setName(name);
+        model.setSize((VirtualMachineSize) vmSizeComboBox.getSelectedItem());
+        model.setUserName(vmUserTextField.getText());
+        model.setPassword(confirmPasswordField.getPassword());
+
+        return wizardStep;
+    }
+
+    @Override
+    public JComponent prepare(WizardNavigationState wizardNavigationState) {
+        rootPanel.revalidate();
+
+        validateEmptyFields();
+
+        final VirtualMachineImage virtualMachineImage = model.getVirtualMachineImage();
+        imageDescriptionTextPane.setText(model.getHtmlFromVMImage(virtualMachineImage));
+        imageDescriptionTextPane.setCaretPosition(0);
+
+        if(vmSizeComboBox.getItemCount() == 0) {
+
+            ProgressManager.getInstance().run(new Task.Backgroundable(project, "Loading VM sizes...", false) {
+                @Override
+                public void run(ProgressIndicator progressIndicator) {
+                    progressIndicator.setIndeterminate(true);
+
+                    try {
+                        final List<VirtualMachineSize> virtualMachineSizes = AzureSDKManagerImpl.getManager().getVirtualMachineSizes(model.getSubscription().getId().toString());
+
+                        Collections.sort(virtualMachineSizes, new Comparator<VirtualMachineSize>() {
+                            @Override
+                            public int compare(VirtualMachineSize t0, VirtualMachineSize t1) {
```

Comment:
The default order on the Azure SDK is alphabetic. The order in the combo is by processors and sizes.